pkg/controller/direct/cloudbuild/externalresource.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudbuild
+
+import (
+	"strings"
+
+	cloudbuildpb "cloud.google.com/go/cloudbuild/apiv1/v2/cloudbuildpb"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/externalresource"
+)
+
+func NewResourceRef(gcpObj *cloudbuildpb.WorkerPool) *ResourceRef {
+	return &ResourceRef{
```

Comment:
This CBWP ResourceRef customize the common ResourceRef, mostly to provide dedicated user information.